plugins/inputs/snmp/gosmi.go

```diff
@@ -34,13 +34,13 @@ type gosmiSnmpTranslateCache struct {
 var gosmiSnmpTranslateCachesLock sync.Mutex
 var gosmiSnmpTranslateCaches map[string]gosmiSnmpTranslateCache
 
-//nolint:revive
-func (g *gosmiTranslator) SnmpTranslate(oid string) (string, string, string, string, error) {
-	a, b, c, d, _, e := g.SnmpTranslateFull(oid)
-	return a, b, c, d, e
+//nolint:revive //function-result-limit conditionally 5 return results allowed
+func (g *gosmiTranslator) SnmpTranslate(oid string) (mibName string, oidNum string, oidText string, conversion string, err error) {
+	mibName, oidNum, oidText, conversion, _, err = g.SnmpTranslateFull(oid)
+	return mibName, oidNum, oidText, conversion, err
```

Comment:
Should we mention this has to be reworked in next major version?

Reply:
I don't think we should. Either someone files an issue and actually works on it or we silently accept it. Scattering TODOs over the code will not help anybody...